link ebtables with --no-as-needed and adjust the link order to fix runtime crash

Program terminated with signal 11, Segmentation fault.
#0  0x00007ffaa837fb53 in ebt_initialize_entry () from /lib64/ebtables/libebtc.so
(gdb) bt
#0  0x00007ffaa837fb53 in ebt_initialize_entry () from /lib64/ebtables/libebtc.so
#1  0x00007ffaa83824dc in do_command () from /lib64/ebtables/libebtc.so
#2  0x000000000040065c in ?? ()
#3  0x00007ffaa7fed755 in __libc_start_main () from /lib64/libc.so.6
#4  0x0000000000400691 in ?? ()

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Index: ebtables-v2.0.10-4/Makefile
===================================================================
--- ebtables-v2.0.10-4.orig/Makefile	2011-12-15 12:02:47.000000000 -0800
+++ ebtables-v2.0.10-4/Makefile	2012-12-17 22:09:45.065973753 -0800
@@ -90,7 +90,7 @@
 	$(CC) -shared $(LDFLAGS) -Wl,-soname,libebtc.so -o libebtc.so -lc $(OBJECTS2)
 
 ebtables: $(OBJECTS) ebtables-standalone.o libebtc.so
-	$(CC) $(CFLAGS) $(CFLAGS_SH_LIB) $(LDFLAGS) -o $@ ebtables-standalone.o -I$(KERNEL_INCLUDES) -L. -Lextensions -lebtc $(EXT_LIBSI) \
+	$(CC) $(CFLAGS) $(CFLAGS_SH_LIB) $(LDFLAGS) -o $@ ebtables-standalone.o -I$(KERNEL_INCLUDES) -L. -Lextensions -Wl,--no-as-needed $(EXT_LIBSI) -lebtc \
 	-Wl,-rpath,$(LIBDIR)
 
 ebtablesu: ebtablesu.c
